cpp/include/cudf/join.hpp

```diff
@@ -438,6 +446,65 @@ class hash_join {
   const std::unique_ptr<impl_type const> _impl;
 };
 
+/**
+ * @brief Unique hash join that builds hash table in creation and probes results in subsequent
+ * `*_join` member functions
+ *
+ * @note Behavior is undefined if the build table contains duplicates.
+ * @note All NaNs are considered as equal
+ *
+ * @tparam HasNested Flag indicating whether there are nested columns in build/probe table
+ */
+// TODO: `HasNested` to be removed via dispatching
```

Comment:
Does this TODO need to be addressed in this PR or a later PR?

Reply:
Probably a separate PR. None of my current solutions is good enough thus need more time to think about this.